rfc/0058-error-handling.md

```diff
@@ -317,6 +317,73 @@ A Note on IDs: The IDs in this RFC are only for organisational purposes and MUST
 
 * Raised when decoding of the data into the user object failed
 
+### 21 RateLimitingFailure
+
+* Raised when a service decides that the caller must be rate limited due to exceeding a rate threshold of some sort.
+* Note that since there are many different reasons why a request is rate limited, the error context MUST include the reason / specific type of rate limiting cause for debugging purposes.
+
+Since the failure itself is a very generic error, an additional Set of `RateLimitingReason` should be part of the error that allows the user - if needed - to further inspect/narrow down the cause. Note that it is a set, since it could contain multiple reasons at the same time.
+
+```
+enum RateLimitingReason {
+  NetworkIngressRateLimitReached,
+  NetworkEgressRateLimitReached,
+  MaximumConnectionsReached,
+  RequestRateLimitReached,
+  ConcurrentRequestLimitReached
+}
+```
+
+Maps to:
+
+* KeyValue
+  * 0x30 RateLimitedNetworkIngress -> NetworkIngressRateLimitReached
+  * 0x31 RateLimitedNetworkEgress -> NetworkEgressRateLimitReached
+  * 0x32 RateLimitedMaxConnections -> MaximumConnectionsReached
+  * 0x33 RateLimitedMaxCommands -> RequestRateLimitReached
```

Comment:
0x34 is the maximum scope data size status. Doesn't appear to be in the documentation yet.